### ISSUE TITLE
Help migrating from theia to code-server

### ISSUE THREAD

**Issue by SDAdham** [eclipse-che/che#17378]
## Summary

Visual studio code is one of the most stable development tool that I've always loved. Since using `Che`, I've been always wishing that `Che` would have VS Code. After seeing `code-server` and that really increased my hopes. Can anyone please really help me migrate from `theia` to `code-server`. I'll be greatly appreciated if I managed to get this done.

## Relevant information

I have checked out `code-server` but I noticed that it doesn't connect to other containers, but I found out there is https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-ssh

I can imaging that integrating `code-server` with `Che` will require a docker image with pre-settings that can get `code-server` integrated with `Che` that would have that remote ssh plugin as mentioned above. See https://github.com/cdr/code-server/issues/830
```
FROM codercom/code-server:latest

COPY extension.vsix /tmp/
RUN code-server --install-extension /tmp/extension.vsix
```

Please help, thanks! Adham

**Comment by SDAdham** [comment]
I did some investigation, https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-ssh is probably not what we need. But here are my thoughts:
* Setup a docker container that would rebuild `cdr/code-server/latest` and add extensions to it.

Those extensions will be those ones that allows you to ssh into other containers using environment variable `$HOSTNAME` to connect the terminal to one of it's containers, the only thing for me is getting the current namespace of where the `che` is running on?

**Comment by SDAdham** [comment]
I created a quick extension https://github.com/SDAdham/vs-code-che for vs code, that would get current workspace and also would show/hide `Che`'s navbar on the left. I didn't test it yet on Eclipse che. I want to rebuild `code-server` image with RUN command `RUN code-server --install-extension /tmp/vs-code-che-0.0.1.vsix`

But not sure how can I get `che` to load my image if I publish the image, from what I noticed in the devfile, if I add image, it would say `Error. Property image is not allowed.`
